### DISCUSSION TITLE
What is currently blocking the release of 0.16.x?  How can we help?

### DISCUSSION THREAD

**Discussion by MailYouLater** [jellyfin/jellyfin-androidtv#2559] [General]
The following things have been stated in discussion #2368:
- The new & improved implementation of the media deletion feature is merged into the codebase, and will be present in the upcoming 0.16.x release cycle, but will not appear in a future (if there are any more) 0.15.x release.
- The 0.16.x release is waiting on some other changes to be ready.

What specifically are these other things which are currently preventing the first of the 0.16.x line from being released? and **how can those of us who are capable of doing so best help to complete these things sooner?**

It was also stated that in the future, the "delete media" feature might be removed from the Jellyfin server, which is ... disconcerting, for users such as those of us who spoke out to request that it be brought back in the Android TV app.  Where can/should we voice our opinion of this where we are likely to be heard? and if this is being considered because of some issue(s) with the current implementation, is there somewhere we should go to help fix/improve it?

And thank you again for all your work on "this huge app".  I am personally enjoying using it, and other than a couple of Live TV related things I'm still trying to get properly set up on the server side, I'm patiently awaiting on the return of the media deletion function before I'll roll it out for the rest of the family to use.  Looking forward to seeing this project continue to improve.  👍

_This is a response to a post by @nielsvanvelzen at https://github.com/jellyfin/jellyfin-androidtv/discussions/2368#discussioncomment-5195676_

Note: That thread got locked due to desrespectful comments and the seemingly combatative attitude of some commenters.  If you are thinking of posting a reply here that could be considered harsh, or even simply unhelpful, please don't.  This is intended to be a discussion of how we, the community can help the project move forward and improve.  Thank you.

**Comment by nielsvanvelzen** [comment]
> What specifically are these other things which are currently preventing the first of the 0.16.x line from being released?

Music playback code rewrite (more info in #1057)

> how can those of us who are capable of doing so best help to complete these things sooner?

At this specific moment there's nothing to do to help with the playback changes until it's in a more refined state. Anybody that wants to help out can still work on translating/reporting issues/helping troubleshoot/fixing issues/other non-playback related stuff.

**Reply by MailYouLater** [reply]
> ... there's nothing to do to help with the playback changes **until it's in a more refined state.**

Any way we can help to get it into the required "more refined state"?

**Reply by nielsvanvelzen** [reply]
No, that's entirely on me

**Reply by MailYouLater** [reply]
... I find this answer very disappointing.  I was really hoping that there was something I could work on that in some way helps get us closer to the next release, and that by extension this might get others interested in helping instead of just complaining...

A response of "that's entirely on me" makes me feel less inclined to even bother, but I'll reread the info in #1057, try to see if I can familiarize myself with the codebase a bit, and see if I can figure out any ways I can help on my own.

**Reply by nielsvanvelzen** [reply]
> I find this answer very disappointing.

Well I can't do anything about that. Software development is not only writing code but also planning and designing. As stated in the linked issue I'm designing the architecture part and slowly submitting code to the upstream repository and I just need to find some hours to finalize it. Dealing with discussions like this one takes up that time.
